plugin/cache.go

```diff
@@ -144,7 +149,13 @@ func (c DuplicateCNameErr) Error() string {
 // querying a specific entry.
 func CachedList(ctx context.Context, g Group) (map[string]Entry, error) {
 	cachedEntries, err := cachedDefaultOp(ctx, ListOp, g, func() (interface{}, error) {
-		entries, err := g.List(ctx)
+		if g.id() == "" {
+			panic("cannot List an entry that you just created")
+		}
+
+		// Include the current entry's ID in context so that operations done while listing children
```

Comment:
I'd elaborate a bit more here, maybe something like "Including the entry's ID allows plugin authors to use any Cached* methods defined on the children _after_ their creation. This is necessary when the child's Cached* methods are used to calculate its attributes. Note that the child's ID is set in cachedOp."

Reason for ^ is because you can still cache operations while listing children, you'd just need to define those operations in the parent's type, which is very annoying.